src/py/mat3ra/made/tools/build/slab/builders.py

```diff
@@ -111,27 +115,32 @@ def _generate(self, configuration: "SlabStrainedSupercellWithGapConfiguration")
         strained_slab_material = super()._generate(configuration)
 
         if configuration.gap is not None:
-            strained_slab_material = self._adjust_lattice_for_gap(strained_slab_material, configuration.gap)
+            strained_slab_material = self._adjust_lattice_for_gap(
+                strained_slab_material, configuration.gap, configuration.gap_direction
+            )
 
         return strained_slab_material
 
-    def _adjust_lattice_for_gap(self, material: Material, gap: float) -> Material:
+    def _adjust_lattice_for_gap(self, material: Material, gap: float, direction: AxisEnum) -> Material:
         """
-        Create a gap in Angstroms in the z-direction of the slab by adjusting the c lattice vector.
-
-        Args:
-            material (Material): The slab material to adjust.
-            gap (float): The gap size in Angstroms to create in the z-direction.
-        Returns:
-            Material: The modified slab material with the adjusted c lattice vector.
+        Adjust the lattice so that the distance from the maximum atomic coordinate in the stacking direction
```

Comment:
Adjust the cell along the stacking direction to make the distance from the cell end to its closest atom to be equal to the gap.